resources/inspec_gem.rb

```diff
@@ -0,0 +1,41 @@
+provides :inspec_gem
+resource_name :inspec_gem
+
+property :gem_name, String, name_property: true
+property :version, String
+property :source, String
+
+default_action :install
+
+action :install do
+  if !version.nil?
+    converge_by "install InSpec version #{version}" do
+      install_inspec_gem(version, source)
+    end
+  elsif !inspec_installed?
+    converge_by 'install latest InSpec version' do
+      install_inspec_gem(nil, source)
+    end
+  else
+    Chef::Log.info("inspec_gem: not installing InSpec. It's already installed and an explicit version was not supplied.")
+  end
+end
+
+action_class do
+  def install_inspec_gem(gem_version, gem_source)
+    chef_gem 'inspec' do
```

Comment:
Do you think it makes sense to remove inspec and train before reinstall? Especially if a version is set, we may get conflicts, since newer versions of the dependencies are not installed.